File: DataStructures/LinkedList/IterativeSearch.java
class Node {

    int data;
    Node  next;
    Node(int d){
        data = d;
        next = null;
    }
}

class LinkedList{
    Node head;


    public void push(int new_key){
        Node new_node = new Node(new_key);
        new_node.next = head;
        head = new_node;
    }


    public bool search(int x)
    {
        Node current = head;
        while(current != null)
        {
            if(current.key == x)
            {
                return true;
            }
            else{
                 current = current.next;
                 return false;
            }
        }
    }
    public static void main(String[] args)
    {
        LinkedList llist = new LinkedList();
        llist.push(2);
        llist.push(21);
        llist.push(3);
        llist.push(45);
        llist.search(llist.head, 21) ? System.out.print("Yes"): System.out.print("No");
    }
} 

File: DataStructures/LinkedList/IterativeSearch.py
class Node:

    def __init__(self, key):
        self.key = key
        self.next = None
class LinkedList:
    def __init__(self):
        self.head = None
    def push(self, new_key):
        new_node = Node(new_key)
        new_node.next = self.head
        self.head = new_node

    def search(self, x):
        current = self.head
        while current != None:
        
            if current.key == x:
                return True
            else:
                current = current.next
                return False
            
        
if __name__ == '__main__':

    llist = LinkedList()

    llist.push(10)
    llist.push(15)
    llist.push(13)
    llist.push(21)

    print("yes") if llist.search(21) else print("no")